### PR TITLE
Cross Chain DAO Tutorial

### PR DESCRIPTION
**Pull request by jboetticher**
### Description

**CURRENTLY IN DRAFT FORM**

Added a tutorial on the smart contracts & design of a cross-chain DAO.  

The tutorial is currently massive and hard to follow, so I'm looking into ways to slim it down and add additional visuals to help explain the architecture and potential alternate architectures better. It's also very rough as I could feel my mind melting near the end.

Feel free to ignore this draft PR if you don't want to provide feedback on it yet, but if you have a lot of time to quickly gloss over it to simply tell me what parts were most confusing, it would be helpful. But don't do it too fast... I need a break from this 😧 

### Checklist

- [ ] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [ ] If images have been added, I have run the `compress-images.py` script to compress the images.
- [X] If this page requires a disclaimer, I have added one

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

**Comment by jboetticher** [comment]
I tried slimming it down a little from its draft version, but it might have ended up longer due to the extra images. This is what I tried to focus on:

- Architecture
- Focus on problems & options
- How a DAO works
- Special parts that are changed in the conversion from single chain to cross chain DAO

This is what I tried to omit:

- Using hardhat
- Testing / deploying

The tutorial is likely still hard to follow, and some things might only make sense to me because I wrote the smart contracts, so feel free to leave critiques on structure and explanations.

File: tutorials/interoperability/cross-chain-dao.md [hunk 1]
```diff
@@ -0,0 +1,829 @@
...
+
+### Cross Chain DAO Contract Part 1 {: #cross-chain-dao-contract-part-1 }
+
+Now to the meat of this tutorial: the cross chain DAO. To be clear, not *all* of the cross-chain logic will be stored in the cross-chain DAO smart contract. Instead, we will separate the hub logic into one contract and the [spoke chain logic into another](#dao-satellite-contract). This makes sense because of the hub-and-spoke model: some of the logic is stored on a single hub chain while the spoke chains interface with it through a simpler satellite contract. We don't need logic meant to be on spoke chains to be on the hub chain.  
+
+We've already got a base for the cross-chain DAO when we used the OpenZeppelin Wizard in a [previous step](#writing-the-cross-chain-dao). It's time to edit it so that it is cross-chain. Let's list out the different functionalities that need to be added:  
+
+1. Support for cross-chain messaging (through LayerZero in this specific example)
+2. A new collection phase between voting and execution
+3. Requesting the collection of votes from spoke chains
```
Comment (eshaben) [inline tutorials/interoperability/cross-chain-dao.md:227]: this refers to the requestCollections function?

File: tutorials/interoperability/cross-chain-dao.md [hunk 2]
```diff
@@ -0,0 +1,829 @@
...
+4. Receiving the collection of votes from spoke chains
```
Comment (eshaben) [inline tutorials/interoperability/cross-chain-dao.md:228]: this refers to the _nonblockingLzReceive function?

File: tutorials/interoperability/cross-chain-dao.md [hunk 3]
```diff
@@ -0,0 +1,829 @@
...
+5. (Optional) Receiving cross-chain messages to do non-voting actions, like proposing or executing
```
Comment (eshaben) [inline tutorials/interoperability/cross-chain-dao.md:229]: this is a part of the _nonblockingLzReceive function?

**Review by eshaben — CHANGES_REQUESTED** [review]

File: tutorials/interoperability/cross-chain-dao.md [hunk 4]
```diff
@@ -0,0 +1,829 @@
...
+
+A good way to play with the configurations of the Governance smart contract is to use the OpenZeppelin smart contract wizard. By going to the [OpenZeppelin contract page](https://www.openzeppelin.com/contracts){target=_blank}, scrolling down, and clicking on the Governance tab, you can view the different ways that you can configure the Governance smart contract. Open it up and play around with it to figure out a simple base for our cross-chain DAO.  
+
+![OpenZeppelin Contract Wizard](/images/tutorials/interoperability/cross-chain-dao/cross-chain-dao-4.png)
+
+We're going to try to keep this base smart contract as simple as possible for demonstration purposes. First, let's name the governor contract to be "CrossChainDAO", since that is what we'll turn it into. Set the voting delay as 0 for simplicity and to make it so that the voting weight snapshot is taken immediately. It is easiest to set the voting period to something short, like 6 minutes. For calculating quorum (the minimum amount of vote weight required for a vote to pass), set it to the number (#) 1. Also disable Timelock, since the timelock period is optional anyways.
```
[the remark this answers is not in the document]
Comment (jboetticher) [reply tutorials/interoperability/cross-chain-dao.md]: Snippet, text, & image didn't properly line up. Fixed in a commit

**Comment by eshaben** [comment]
@albertov19 this is ready for you to review now if you wanna take a look 👀

**Review by eshaben — APPROVED** [review]
LGTM

File: tutorials/interoperability/cross-chain-dao.md [hunk 5]
```diff
@@ -0,0 +1,884 @@
...
+
+In fact, the `_countVote` function is [directly copied](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4fb6833e325658946c2185862b8e57e32f3683bc/contracts/governance/extensions/GovernorCountingSimple.sol#L78){target=_blank} from the hub chain! Much of the logic of single-chain dApps can be reused in cross-chain dApps with minor tweaks.  
+
+That's it for breaking down the satellite contract. It was more or less simple because most of the logic is just a reflection of what happens on the hub chain. You can view the completed smart contract in its [GitHub repository](https://github.com/jboetticher/cross-chain-dao/blob/main/contracts/DAOSatellite.sol){target=_blank}.  
+
+At this point, every single smart contract has been finished, and a deployment scheme like the one below can be made. If you are interested in seeing this in action, the [GitHub repository](https://github.com/jboetticher/cross-chain-dao){target=_blank} that hosts the cross-chain DAO allows you to deploy on TestNets.  
+
+![Smart contracts overview](/images/tutorials/interoperability/cross-chain-dao/cross-chain-dao-3.png)  
+
+## Caveats and Other Designs {: #caveats-and-other-designs }
```
Comment (albertov19) [inline tutorials/interoperability/cross-chain-dao.md:808]: Maybe add here another disclaimer about the code not being meant for production but educational?

**Review by albertov19 — CHANGES_REQUESTED** [review]
LGTM! 

@eshaben do you feel we should organize the left hand side navigation to be alphabetical? If so, connected contracts should be above XCM

**Comment by albertov19** [comment]
Also I did do a commit, please review :)
